test/e2e/external_oidc_test.go

```diff
@@ -187,263 +290,460 @@ func newTestClient(t *testing.T) (*testClient, error) {
+						Name: caBundleName,
+					},
+				},
+				ClaimMappings: configv1.TokenClaimMappings{
+					Username: usernameMapping,
+					Groups: configv1.PrefixedClaimMapping{
+						TokenClaimMapping: configv1.TokenClaimMapping{
+							Claim: oidcGroupsClaim,
+						},
+					},
+				},
+			},
+		},
 	}
 
-	tc.t.Log("will sync OIDC ca-bundle to KAS static pod resources")
+	return auth
+}
+
+func (tc *testClient) validateKASConfig(t *testing.T, ctx context.Context) int32 {
+	kas, err := tc.operatorConfigClient.OperatorV1().KubeAPIServers().Get(ctx, "cluster", metav1.GetOptions{})
+	require.NoError(t, err)
 
-	signingKey, err := tc.kubeClient.CoreV1().ConfigMaps(oidcCABundleConfigMapNamespace).Get(ctx, oidcCABundleConfigMap, metav1.GetOptions{})
-	if err != nil {
-		return
+	var observedConfig map[string]any
+	err = json.Unmarshal(kas.Spec.ObservedConfig.Raw, &observedConfig)
+	require.NoError(t, err)
+
+	apiServerArguments := observedConfig["apiServerArguments"].(map[string]any)
+
+	require.Nil(t, apiServerArguments["authentication-token-webhook-config-file"])
+	require.Nil(t, apiServerArguments["authentication-token-webhook-version"])
+	require.Nil(t, observedConfig["authConfig"])
+
+	authConfigArg := apiServerArguments["authentication-config"].([]any)
+	require.NotEmpty(t, authConfigArg)
+	require.Equal(t, authConfigArg[0].(string), "/etc/kubernetes/static-pod-resources/configmaps/auth-config/auth-config.json")
+
+	return kas.Status.LatestAvailableRevision
+}
+
+func (tc *testClient) validateAuthConfigJSON(t *testing.T, ctx context.Context, idpURL string, authSpec *configv1.AuthenticationSpec, usernamePrefix string, kasRevision int32) {
+	caBundleName := authSpec.OIDCProviders[0].Issuer.CertificateAuthority.Name
+	certData := ""
+	if len(caBundleName) > 0 {
+		cm, err := tc.kubeClient.CoreV1().ConfigMaps("openshift-config").Get(ctx, caBundleName, metav1.GetOptions{})
+		require.NoError(t, err)
+		certData = cm.Data["ca-bundle.crt"]
 	}
 
-	// This functionality requires defining "oidc-serving-ca" as a RevisionConfigMap at the KAS-o
-	// See example PR: https://github.com/openshift/cluster-kube-apiserver-operator/pull/1720
-	oidcServingCA := &corev1.ConfigMap{
-		ObjectMeta: metav1.ObjectMeta{
-			Name:      "oidc-serving-ca",
-			Namespace: kasNamespace,
-		},
-		Data: map[string]string{
-			"ca-bundle.crt": string(signingKey.Data["tls.crt"]),
-		},
+	expectedAuthConfigJSON := fmt.Sprintf(`{"kind":"AuthenticationConfiguration","apiVersion":"apiserver.config.k8s.io/v1beta1","jwt":[{"issuer":{"url":"%s","certificateAuthority":"%s","audiences":[%s],"audienceMatchPolicy":"MatchAny"},"claimMappings":{"username":{"claim":"%s","prefix":"%s"},"groups":{"claim":"%s","prefix":"%s"},"uid":{}}}]}`,
+		idpURL,
+		strings.ReplaceAll(certData, "\n", "\\n"),
+		strings.Join([]string{fmt.Sprintf(`"%s"`, oidcClientId)}, ","),
+		authSpec.OIDCProviders[0].ClaimMappings.Username.Claim,
+		usernamePrefix,
+		oidcGroupsClaim,
+		oidcGroupsPrefix,
```

Comment:
Don't mix global variables and function/method parameters where you can avoid it. IMO it causes unnecessary confusion when looking at what a particular function/method does.